sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java

```diff
@@ -0,0 +1,681 @@
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.io.hbase.coders.HBaseMutationCoder;
+import org.apache.beam.sdk.io.hbase.coders.HBaseResultCoder;
+import org.apache.beam.sdk.io.hbase.coders.SerializableConfiguration;
+import org.apache.beam.sdk.io.hbase.coders.SerializableScan;
+import org.apache.beam.sdk.io.range.ByteKeyRange;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.hadoop.conf.Configuration;
+
+import org.apache.hadoop.hbase.ClusterStatus;
+import org.apache.hadoop.hbase.HRegionLocation;
+import org.apache.hadoop.hbase.RegionLoad;
+import org.apache.hadoop.hbase.ServerLoad;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.BufferedMutator;
+import org.apache.hadoop.hbase.client.BufferedMutatorParams;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.RegionLocator;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A bounded source and sink for HBase.
+ *
+ * <p>For more information, see the online documentation at
+ * <a href="https://hbase.apache.org/">HBase</a>.
+ *
+ * <h3>Reading from HBase</h3>
+ *
+ * <p>The HBase source returns a set of rows from a single table, returning a
+ * {@code PCollection<Result>}.
+ *
+ * <p>To configure a HBase source, you must supply a table id and a {@link Configuration}
+ * to identify the HBase instance. By default, {@link HBaseIO.Read} will read all rows in the
+ * table. The row range to be read can optionally be restricted using with a {@link Scan} object
+ * or using the {@link HBaseIO.Read#withKeyRange}, and a {@link Filter} using
+ * {@link HBaseIO.Read#withFilter}, for example:
+ *
+ * <pre>{@code
+ * // Scan the entire table.
+ * p.apply("read",
+ *     HBaseIO.read()
+ *         .withConfiguration(configuration)
+ *         .withTableId("table"));
+ *
+ * // Filter data using a HBaseIO Scan
+ * Scan scan = ...
+ * p.apply("read",
+ *     HBaseIO.read()
+ *         .withConfiguration(configuration)
+ *         .withTableId("table"))
+ *         .withScan(scan));
+ *
+ * // Scan a prefix of the table.
```

Comment:
I copy exactly this lined from the BigtableIO doc, tell me what you prefer and I will change it.
https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L102